api/python/setup.py

```diff
@@ -56,7 +56,7 @@ def run(self):
     url='https://github.com/quiltdata/quilt',
     keywords='',
     install_requires=[
-        'appdirs>=1.4.0',
+        'platformdirs>=2.5.2',
```

Comment:
Makes sense?
 ```suggestion
        'platformdirs>=2,<3',
```

Reply:
I don’t think so. If you’d add a comment here that explains why you add an upper limit, maybe, but you shouldn’t do it without a reason: https://iscinumpy.dev/post/bound-version-constraints/

Reply:
OK, I don't insist on setting upper limit, but any reason for setting lower limit to the newest version? Doesn't 'platformdirs==2.0.0' really works for us?

Reply:
Definitely not, you don’t want to pin any dependencies in a library.

You might want to read the “Python is not JavaScript” section of the linked article.

Reply:
"`>=` everywhere" doesn't seem safe tbh, bc it means potentially accepting some version with breaking changes in the future, especially considering that python community starts to embrace SemVer (well, yeah, somewhat).
but i think what @sir-sigurd meant in this particular case is actually `platformdirs>=2.0.0` (more relaxed lower bound) vs having a lower bound of `2.5.2`

Reply:
`platformdirs>=2` would work, yes. I often set pins to the current version because

1. I’m not testing with lower versions, so I don’t know if I’m relying on a bug fix or a feature of the current version, and can therefore only say that that one works. If someone files an issue in my project wanting to use some lower version for some reason I happily lower it. Another reason is that
2. Some libraries *do* pin dependencies or specify upper bounds (mostly because they don’t know better, sometimes for good reasons). In that case `pip` can sometimes end up veeery slooowly downloading a bunch of versions of many packages to maybe find a compatible combination. Setting your lower bounds relatively high means the space it searches is smaller resulting in a quicker resolution (or notification of resolution failing) when that happens.

I saw that you’re doing some pinning in your dependencies. We should fix that so you don’t cause the problem I described for people. See also https://iscinumpy.dev/post/bound-version-constraints/#python-is-not-javascript

Reply:
> ">= everywhere" doesn't seem safe tbh, bc it means potentially accepting some version with breaking changes in the future

Please read the blog post if you think that. Henry and I know the Python packaging landscape quite well, and this is an informed decision that takes your concerns into account.

I know it’s long, but I’m sure you can Ctrl-F your way to the parts that address your questions.